aptos-move/framework/aptos-framework/sources/AuctionHouse.move

```diff
@@ -0,0 +1,878 @@
+// This module provides a basic Auction house with first-price auction
+
+module AptosFramework::AuctionHouse {
+    use AptosFramework::Coin::{Self, Coin};
+    use AptosFramework::IterableTable::{Self, IterableTable};
+    use AptosFramework::Table::{Self, Table};
+    use AptosFramework::TestCoin::TestCoin;
+    use AptosFramework::Timestamp;
+    use AptosFramework::Token::{Self, Token, TokenId};
+    use Std::Event::{Self, EventHandle};
+    use Std::GUID::{Self, ID};
+    use Std::Option::{Self, Option};
+    use Std::Signer;
+    use Std::Vector;
+
+    // Auction has 4 states:
+    // 1/ Active: current_time >= start_time and current_time <= start_time + duration
+    // 2/ Inactive: current_time < start_time or current_time > start_time + duration
+    // 3/ Completed: current_time > start_time + duration
+    // 4/ Unclaimed: Completed and (token not claimed or coins not claimed)
+
+    const EINVALID_BUYER: u64 = 0;
+    const EINSUFFICIENT_BID: u64 = 1;
+    const ELISTING_INACTIVE: u64 = 2;
+    const EAUCTION_NOT_COMPLETE: u64 = 3;
+    const ENOT_CLAIMABLE: u64 = 4;
+    const ETOKEN_INVOLVED_IN_ANOTHER_AUCTION: u64 = 5;
+    const EALREADY_CLAIMED: u64 = 6;
+    const EINVALID_TOKEN_ID: u64 = 7;
+    const EINSUFFICIENT_MIN_SELLING_PRICE: u64 = 8;
+    const EBUY_NOW_UNAVAILABLE: u64 = 9;
+
+    // Represents an auction
```

Comment:
have a little more description to help people understand
use `/// comment ` will help auto generate the doc